### DISCUSSION TITLE
receiving messages on rmb-remote entry point where $dst != $local

### DISCUSSION THREAD

**Discussion by sameh-farouk** [threefoldtech/rmb-rs#5] [General]
> RMB (remote)
On receiving a message on the /rmb-remote entrypoint. This entry-point will verify message body, and signature. A 202 Accepted is returned if all is okay, 400 BadRequest, or 401 Unauthorized if signature verifications fails.
If message $dst == $local, $ret is set to msgbus.system.reply and pushed to msgbus.$cmd

![Screenshot from 2022-04-07 12-19-04](https://user-images.githubusercontent.com/42457449/162177752-c2e42465-16ca-40e1-8350-25052ec772da.png)

in the chart, HTTP handler (on /rmb-remote entry point) would push the message to `msgbus.system.forward' on cases where $dst != $local, but not clear how/when this would happen (receiving a message on /rmb-remote that intended to be sent to another twin). I believe RMB should return an error response if this happened, otherwise, what is the successful HTTP response that would be sent back upon receiving this message from the (intermediary) RMB and before forwarding it? is that part of an alternative flow to the current proxy flow. would be great to clear these points.

**Comment by muhamadazmy** [comment]
This should not happen (normally) but i was wondering if supporting `proxing` messages should be implemented. The receiver rmb still can process the message, and then answer to the original sender directly (since he is still the source?) 

Anyway, i think you are right about not supporting this. it's going to be very confusing and not sure if this will work even with the signing. 
Let's drop that
